File: src/test/java/interview/reader/FileInputReaderTest.java
package interview.reader;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

/**
 * @author : Xenofon Zinoviou
 */
class FileInputReaderTest {

    private FileInputReader testClass;

    @BeforeEach
    void init() {
        testClass = new FileInputReader();
    }

    @Test
    void test_givenValidFile_shouldReturnFileContentsAsString() {
        String expected = "the\na\non\noff\n ";
        final String output = testClass.readInput();
        Assertions.assertEquals(expected, output);
    }


}

File: src/test/kotlin/inputReader/FileInputReaderTest.kt
package inputReader

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class FileInputReaderTest {

    @Test
    fun readReturnsTheWholeContentOfTheProvidedFile() {
        // given
        val textFilePath = "src/test/resources/NormalInputText.txt"
        val reader = FileInputReader(textFilePath)

        // when
        val result = reader.read()

        // then
        val expected = listOf("Mary had", "a little", "lamb").joinToString("\n")
        assertEquals(expected, result)
    }

    @Test
    fun readReturnsEmptyStringInCaseOfEmptyFile() {
        val textFilePath = "src/test/resources/EmptyInputText.txt"
        val reader = FileInputReader(textFilePath)

        // when
        val result = reader.read()

        // then
        val expected = ""
        assertEquals(expected, result)
    }

    @Test
    fun readReturnsEmptyStringInCaseOfBlankFile() {
        val textFilePath = "src/test/resources/BlankInputText.txt"
        val reader = FileInputReader(textFilePath)

        // when
        val result = reader.read()

        // then
        val expected = ""
        assertEquals(expected, result)
    }
}